src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -766,11 +766,19 @@ def set_operator(
         return new
 
     def limit(self, n: int, *, offset: int = 0) -> "SelectStatement":
-        new = copy(self)
-        new.from_ = self.from_.to_subqueryable()
-        new.limit_ = min(self.limit_, n) if self.limit_ else n
-        new.offset = (self.offset + offset) if self.offset else offset
-        new.column_states = self.column_states
+        if offset and self.limit_:  # Don't flatten when there both layers have offset.
```

Comment:
The documentation is suggesting `if offset and self.offset`? Shall we modify the documentation?

Reply:
Good catch. Updated the comments.